ocaml-lsp-server/src/merlin_config.ml

```diff
@@ -390,8 +392,12 @@ let config (t : t) : Mconfig.t Fiber.t =
           use_entry entry
     in
     let+ dot, failures = get_config entry.process ~workdir:ctx.workdir t.path in
-    let merlin = Config.merge dot t.initial.merlin failures config_path in
-    Mconfig.normalize { t.initial with merlin }
+
+    if !should_read_dot_merlin && dot = Config.empty then
+      Mconfig.get_external_config t.path t.initial
```

Comment:
Is it ok not to normalize here?

Reply:
I looked at the [source](https://github.com/rgrinberg/merlin/blob/e757fe8f9276c87154356171959fabc806f4488e/src/kernel/mconfig.ml#L232) and `get_external_config` already calls normalize.